boot/header.S

```diff
@@ -1,11 +1,31 @@
 // SPDX-License-Identifier: GPL-2.0
 //
-// header.S supports booting directly from a UEFI BIOS or via an intermediate
-// bootloader that supports the Linux boot protocol. When booted directly from
-// the BIOS, it provides the MS-DOS & PE/COFF headers. When using an intermediate
-// bootloader, it provides the first few bytes of the Linux boot header (at the
-// end of the boot sector), with the remainder of the header being provided by
-// setup.S.
+// header.S supports booting directly from the 1) BIOS (FDD), 2) UEFI or 3) via
+// an intermediate bootloader that supports the Linux boot protocol. All-in-one.
+//
+// 1) If booted directly from the BIOS, ".bootsect" gets loaded at address 0x7C00
+// with DL register set to the boot drive number: 0x00 - FDD, 0x80 - first HDD.
+// Unfortunately, when executing "MZ" signature as x86 instructions, the original
+// value of DX becomes instantly and irrevocably lost, but this is fine as we only
+// support booting from FDD anyway. Once started, the code immediately prints
+// boot_msg - "Loading ". Next, it sets DX to 0x00 (as assumed by the code) and
+// creates a new Floppy Disk Drive parameter table. Then, it tries to load
+// ".setup" section immediately after itself (address 0x7E00) that contains
+// setup code, Once succesfull, it prints memtest version and starts loading
+// the main program code at address 0x10000. After each successfull read,
+// a dot - "." is printed. If read fails, error message gets printed and
+// the read operation is restarted. At the end, 07E0h:0000h is called which
+// immediately jumps into do_setup that handles switches CPU into protected mode
+// and calls kernel entry point at 0x10000.
+//
+// 2) To support loading from the UEFI, it provides the MS-DOS & PE/COFF headers.
+// The PE/COFF headers point to offset 0x10000 + 0x1E0 where efi_boot from
+// startup32.S / startup64.S that handles EFI handover and the rest of booting.
+//
+// 3) When using an intermediate bootloader, it provides the first few bytes of
+// the Linux boot header (at the end of the ".bootsect" section), with the
+// remainder of the header and the setup code being provided in the ".setup".
+// section.
 //
 // Copyright (C) 2020-2023 Martin Whitaker.
```

Comment:
Update copyright to 20234

Reply:
Done

Reply:
Actually https://github.com/memtest86plus/memtest86plus/commit/e38ad8e6a406ef7a5fd6cdfcbf2c4ab5859cb2af already took care of this?

Reply:
Ah, good point!